libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt

```diff
@@ -55,7 +55,11 @@ class MapboxOffboardRouter(
             }
 
             override fun onFailure(call: Call<DirectionsResponse>, t: Throwable) {
-                callback.onFailure(t)
+                if (call.isCanceled) {
```

Comment:
From what I've seen, if the request is canceled mid-flight it will always report `onFailure`. We can add the same check above as well though, it won't hurt.

Reply:
Can the `if-else` be changed to use `when`?